pkg/server/plugin/keymanager/awskms/awskms.go

```diff
@@ -707,6 +744,75 @@ func (p *Plugin) notifyDisposeKeys(err error) {
 	}
 }
 
+func (p *Plugin) createDefaultPolicy(ctx context.Context) (*string, error) {
+	result, err := p.stsClient.GetCallerIdentity(ctx, &sts.GetCallerIdentityInput{})
+	if err != nil {
+		return nil, fmt.Errorf("cannot get caller identity: %w", err)
+	}
+
+	accountID := *result.Account
+	roleName, err := roleNameFromARN(*result.Arn)
+	if err != nil {
+		// the server has not assumed any role, use default KMS policy and log a warn message
+		p.log.Warn("In a future version of SPIRE, it will be mandatory for the SPIRE servers to assume an AWS IAM Role when using the default AWS KMS key policy. Please assign an IAM role to this SPIRE Server instace.")
+		return nil, nil
+	}
```

Comment:
How likely is this to happen? Due to the placement of `createDefaultPolicy` this will be attempted, and long,  every time we create a key, right? Maybe that's ok?

Reply:
I'm convinced this is ok.